fix(plugin-commands-patching): exclude .DS_Store file at patch-commit

huuyafwww opened the pull request:
fix #8922

huuyafwww commented on patching/plugin-commands-patching/src/patchCommit.ts:
```diff
@@ -165,6 +165,8 @@ async function diffFolders (folderA: string, folderB: string): Promise<string> {
     .replace(new RegExp(escapeStringRegexp(`${folderAN}/`), 'g'), '')
     .replace(new RegExp(escapeStringRegexp(`${folderBN}/`), 'g'), '')
     .replace(/\n\\ No newline at end of file\n$/, '\n')
+    .replace(/^diff --git a\/.*\.DS_Store b\/.*\.DS_Store[\s\S]*?(?=^diff --git)/gm, '')
```
The cause was that the --no-index option caused .gitignore to be ignored.
However, the --no-index option cannot be turned off because the change was made within node_modules.
Also, I tried to exclude certain files with the git diff command, but that didn't work either, so I fixed it with a regular expression

```js
await execa('git', ['-c', 'core.safecrlf=false', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--ignore-cr-at-eol', '--irreversible-delete', '--full-index', '--no-index', '--text', '--no-ext-diff', folderAN, folderBN]
```

This regex corresponds to the difference except for the tail in PATCH.

huuyafwww commented on patching/plugin-commands-patching/src/patchCommit.ts:
```diff
@@ -165,6 +165,8 @@ async function diffFolders (folderA: string, folderB: string): Promise<string> {
     .replace(new RegExp(escapeStringRegexp(`${folderBN}/`), 'g'), '')
     .replace(/\n\\ No newline at end of file\n$/, '\n')
+    .replace(/^diff --git a\/.*\.DS_Store b\/.*\.DS_Store[\s\S]*?(?=^diff --git)/gm, '')
+    .replace(/^diff --git a\/*?\.DS_Store b\/*?\.DS_Store[\s\S].*$/gsm, '')
```
This regex corresponds to the case where the difference of DS_Store exists at the end in PATCH.

zkochan reviewed (commented):
can we cover it with a test? Maybe a unit test will be enough that just checks the regex. 

Also, as an alternative we could just remove the `.DS_Store` directories before diffing, no?

huuyafwww commented:
@zkochan 
Thanks for checking this PR.
The regular expression was incorrect, so we corrected it and added a test.

> Also, as an alternative we could just remove the .DS_Store directories before diffing, no?

The .DS_Store contains meta information about files and folders.
DS_Store is a file that is generated by itself in the MacOS environment,
It is also an invisible file that cannot be seen by normal applications.
It is a file managed by the OS, not the user.
Therefore, we prefer to avoid deleting it if at all possible,
This approach has been used to correct the problem.
